includes/ziti/ziti_buffer.h

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (c) 2021 NetFoundry Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ZITI_SDK_ZITI_BUFFER_H
+#define ZITI_SDK_ZITI_BUFFER_H
+
+#include <stdint.h>
+#include "externs.h"
+#include "ziti_log.h"
+
+#if !defined(__DEFINED_ssize_t) && !defined(__ssize_t_defined)
+#if _WIN32
+typedef intptr_t ssize_t;
+#define __DEFINED_ssize_t
+#define __ssize_t_defined
+#else
+#include <unistd.h>
+#endif
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Growing string buffer.
+ *
+ * string_buf allocates memory internally as needed unless it was created with [new_fixed_string_buf()]
+ */
+typedef struct string_buf_s string_buf_t;
+
+/**
+ * Create new string buffer.
+ * @return new buffer instance
+ */
+ZITI_FUNC string_buf_t *new_string_buf();
```

Comment:
no tests for: new_string_buf, new_fixed_string_buf, delete_string_buf, string_buf_append_byte